code/game/gamemodes/cult/cult_items.dm

```diff
@@ -682,3 +682,105 @@
+			return
+		var/obj/effect/rune/teleport/R = O
+		attempt_portal(R, user)
+
+/obj/item/portal_amulet/proc/attempt_portal(obj/effect/rune/teleport/R, mob/user)
+	var/list/potential_runes = list()
+	var/list/teleportnames = list()
+	var/list/duplicaterunecount = list()
+	var/turf/T = get_turf(src) //used to tell the other rune where we came from
+
+	for(var/I in GLOB.teleport_runes)
+		var/obj/effect/rune/teleport/target = I
+		var/resultkey = target.listkey
+		if(resultkey in teleportnames)
+			duplicaterunecount[resultkey]++
+			resultkey = "[resultkey] ([duplicaterunecount[resultkey]])"
+		else
+			teleportnames += resultkey
+			duplicaterunecount[resultkey] = 1
+		if(target != R && is_level_reachable(target.z))
+			potential_runes[resultkey] = target
+
+	if(!length(potential_runes))
+		to_chat(user, "<span class='warning'>There are no valid runes to teleport to!</span>")
+		return
+
+	if(!is_level_reachable(user.z))
+		to_chat(user, "<span class='cultitalic'>You are not in the right dimension!</span>")
+		return
+
+	var/input_rune_key = input(user, "Choose a rune to make a portal to.", "Rune to make a portal to") as null|anything in potential_runes //we know what key they picked
+	var/obj/effect/rune/teleport/actual_selected_rune = potential_runes[input_rune_key] //what rune does that key correspond to?
+	if(QDELETED(R) || QDELETED(actual_selected_rune) ||!Adjacent(user) || user.incapacitated())
+		return
+
+	if(is_mining_level(R.z) && !is_mining_level(actual_selected_rune.z))
+		actual_selected_rune.handle_portal("lava")
+	else if(!is_station_level(R.z) || istype(get_area(src), /area/space))
+		actual_selected_rune.handle_portal("space", T)
+	new /obj/effect/portal/cult(get_turf(R), get_turf(actual_selected_rune), src, 4 MINUTES)
+	to_chat(user, "<span class='cultitalic'>You use the magic of the amulet to turn [R] into a portal.</span>")
+	playsound(src, 'sound/magic/cult_spell.ogg', 100, TRUE)
+	qdel(R)
+	qdel(src)
+
+/obj/effect/portal/cult
+	name = "eldritch portal"
+	desc = "An evil portal made by dark magics. Suprisingly stable."
+	icon_state = "portal1"
+	failchance = 0
+	precision = FALSE
+	var/obj/effect/temp_visual/cult_portal_exit/exit = null
+
+/obj/effect/portal/cult/New()
+	..()
+	exit = new /obj/effect/temp_visual/cult_portal_exit
+	exit.forceMove(target)
+
+/obj/effect/portal/cult/attackby(obj/I, mob/user, params)
+	if(istype(I, /obj/item/melee/cultblade/dagger) && iscultist(user) || istype(I, /obj/item/nullrod) && user.mind.isholy)
+		to_chat(user, "<span class='notice'>You close the portal with your [I].</span>")
+		playsound(src, 'sound/magic/magic_missile.ogg', 100, TRUE)
+		qdel(src)
+	return ..()
+
+/obj/effect/portal/cult/Destroy()
+	qdel(exit)
+	return ..()
+
+/obj/effect/temp_visual/cult_portal_exit
+	name = "eldritch rift"
+	desc = "An exit point for some cult portal. Be on guard, more things may come out of it"
+	icon = 'icons/obj/biomass.dmi'
+	icon_state = "rift"
+	duration = 4 MINUTES
```

Comment:
Minor thing but this is hard tied to the `attempt_portal` at the moment. If the `/obj/effect/portal/cult` ever gets made by something else for like 5 minutes then the exit will be deleted before the entrance gets deleted

Reply:
Changed it from a temp visual